src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/BindingFlagSupport/QueryResult.cs

```diff
@@ -112,11 +112,10 @@ public void CopyTo(MemberInfo[] array, int startIndex)
                         // Assuming the policy says it's ok to ignore the ambiguity, we're to resolve in favor of the member
                         // declared by the most derived type. Since QueriedMemberLists are sorted in order of decreasing derivation,
                         // that means we let the first match win - unless, of course, they're both the "most derived member".
-                        if (match.DeclaringType.Equals(challenger.DeclaringType))
-                            throw new AmbiguousMatchException();
-
-                        if (!_policies.OkToIgnoreAmbiguity(match, challenger))
-                            throw new AmbiguousMatchException();
+                        // If they're not the same method, we throw if the policy doesn't allow ambiguity.
```

Comment:
It doesn't seem like for method only, in general the 1st part of the sentence doesn't make sense to me.
```suggestion
                        // If they're not from same type, we throw if the policy doesn't allow ambiguity.
```